app/src/components/submissionPage/SnapList.tsx

```diff
@@ -64,12 +59,40 @@ const SnapList: React.FunctionComponent<SnapListProps> = ({ snaps }) => {
         });
     }
 
+    function getNameFromTag(tag: string): string {
+        const namePart = tag.match(fullNameRegex)[0];
+        return namePart.substring(1, namePart.length - 1);
+    }
+
+    const formattedSnapBody = (body: string) => {
+        const tags = body.match(tagRegex);
+        if (!tags) {
+            return <span>{body}</span>;
+        }
+        const arbitraryDelimiter = "--@!--";
+        const withDelimters = body.replace(
+            tagRegex,
+            `${arbitraryDelimiter}$&${arbitraryDelimiter}`
+        );
+        const split = withDelimters.split(arbitraryDelimiter);
```

Comment:
JavaScript can split strings based on regular expressions, which should let you avoid the arbitraryDelimiter.